### PR TITLE
update_iface_link_state: wait for guest network reset

### PR DESCRIPTION
**Pull request by smitterl**
On our nested systems backed by a virtual NAT network it takes longer for the guest to get a reachable network.

The test was failing at the ping intent with

ping: connect: Network is unreachable

Instead, retry for up to 5 seconds to succeed.

### CODE REVIEW

**Comment by smitterl** [comment]
@chloerh @yanqzhan Please help review

**Comment by smitterl** [comment]
```
JOB ID     : e279280bcfaa1bfad1c53c58730dcd41f3e99fdc
JOB LOG    : /var/log/avocado/job-results/job-2024-07-29T09.48-e279280/job.log
 (1/5) type_specific.io-github-autotest-libvirt.virtual_network.update_device.iface_link_state.direct.virtio.add: PASS (28.48 s)
 (2/5) type_specific.io-github-autotest-libvirt.virtual_network.update_device.iface_link_state.direct.virtio.delete: PASS (32.97 s)
 (3/5) type_specific.io-github-autotest-libvirt.virtual_network.update_device.iface_link_state.ethernet.virtio.update: PASS (49.45 s)
 (4/5) type_specific.io-github-autotest-libvirt.virtual_network.link_state.model.direct.virtio.initial_up: PASS (31.19 s)
 (5/5) type_specific.io-github-autotest-libvirt.virtual_network.update_device.update_iface_trustGuestRxFilters.no_to_yes: PASS (33.15 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/log/avocado/job-results/job-2024-07-29T09.48-e279280/results.html
JOB TIME   : 175.80 s

```

**Comment by smitterl** [comment]
```
JOB ID     : e279280bcfaa1bfad1c53c58730dcd41f3e99fdc
JOB LOG    : /var/log/avocado/job-results/job-2024-07-29T09.48-e279280/job.log
 (1/5) type_specific.io-github-autotest-libvirt.virtual_network.update_device.iface_link_state.direct.virtio.add: PASS (28.48 s)
 (2/5) type_specific.io-github-autotest-libvirt.virtual_network.update_device.iface_link_state.direct.virtio.delete: PASS (32.97 s)
 (3/5) type_specific.io-github-autotest-libvirt.virtual_network.update_device.iface_link_state.ethernet.virtio.update: PASS (49.45 s)
 (4/5) type_specific.io-github-autotest-libvirt.virtual_network.link_state.model.direct.virtio.initial_up: PASS (31.19 s)
 (5/5) type_specific.io-github-autotest-libvirt.virtual_network.update_device.update_iface_trustGuestRxFilters.no_to_yes: PASS (33.15 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/log/avocado/job-results/job-2024-07-29T09.48-e279280/results.html
JOB TIME   : 175.80 s
```

File: provider/virtual_network/network_base.py [hunk 1]
```diff
@@ -110,15 +111,23 @@ def ping_check(params, ips, session=None, force_ipv4=True, **args):
...
+            status, output = utils_net.ping(dest=dest_ip, count=5,
+                                            timeout=10,
+                                            session=ping_session,
+                                            force_ipv4=force_ipv4,
+                                            **ping_args)
+            if status and "unreachable" in output and expect_result == 'pass':
+                return False
+            return status == 0
+
+        status = utils_misc.wait_for(_ping, timeout=10)
```
[the remark this answers is not in the document]
Comment (smitterl) [reply provider/virtual_network/network_base.py]: You're right, thank you. Updated to 15 to make sure it's tried a second time.

**Review by chloerh — APPROVED** [review]

**Review by chunfuwen — APPROVED** [review]
lgtm

**Pull request merged**